pythia/seo/ELO.py

```diff
@@ -0,0 +1,195 @@
+                                                self.rankings.loc[image_0]['score']) / 400.00))
+        return expexted_0_score
+
+    def new_rating(self, image, score_for_image, image_expected_score):
+        """
+        Calculates new rating based on the ELO algorithm.
+
+        Parameters
+        ----------
+        image : int
+            Image id for the image
+        score_for_image : int
+            Actual result of classification of the image in a pairwise match.
+            `0` denotes less complex, `1` denotes more complex
+        image_expected_score : float
+            Expected result of classification of image in a pairwise match
+            based on current rating of the image.
+
+        Returns
+        -------
+        new_image_rating : float
+            New rating of image after the classification match.
+        """
+        new_image_rating = self.rankings.loc[image]['score'] + self.rankings.loc[image]['k_value'] * (score_for_image - image_expected_score)
+        return new_image_rating
+
+    def score_update(self, image_0, image_1, score_for_image_0):
+        """
+        Updates the ratings of the two images based on the complexity classification.
+
+        Parameters
+        ----------
+        image_0 : int
+            Image id for first image
+        image_1 : int
+            Image id for second image
+        score_for_image_0 : int
+            Actual result of classification of the image 0 in a pairwise match.
+            `0` denotes less complex, `1` denotes more complex
+
+        Notes
+        -----
+        To make updates in the original rankings DataFrame, for each classification,
+        two state dictionaries need to be maintained, corresponfing to the two AR images.
+        The changes are made to these state dictionaries and then the ranking DataFrame is updated.
+        """
+        # state dicts
+        state_dict_0 = self.rankings.loc[image_0].to_dict()
+        state_dict_0['last_scores'] = list(map(float, state_dict_0['last_scores'].split(',')))
+        state_dict_1 = self.rankings.loc[image_1].to_dict()
+        state_dict_1['last_scores'] = list(map(float, state_dict_1['last_scores'].split(',')))
+
+        expected_score_0 = self.expected_score(image_0, image_1)
+        expected_score_1 = 1 - expected_score_0
+
+        new_rating_0 = self.new_rating(image_0, score_for_image_0, expected_score_0)
+        new_rating_1 = self.new_rating(image_1, 1 - score_for_image_0, expected_score_1)
+
+        state_dict_0['last_scores'].append(new_rating_0)
+        state_dict_1['last_scores'].append(new_rating_1)
+
+        if len(state_dict_0['last_scores']) > self.score_memory:
+            state_dict_0['last_scores'].pop(0)
+        if len(state_dict_1['last_scores']) > self.score_memory:
+            state_dict_1['last_scores'].pop(0)
+
+        new_std_dev_0 = min(np.std(state_dict_0['last_scores']), 1000000)  # prevents Infinity
```

Comment:
```suggestion
        new_std_dev_0 = min(np.std(state_dict_0['last_scores']), 1_000_000)  # prevents Infinity
```
To ease readability of this "magic number". And since it's a magic number, then probably better to assign it to a variable.

Reply:
I did not know we could do this in python!